ECommerceDataProvider/Controllers/DataProviderController.cs

```diff
@@ -0,0 +1,51 @@
+﻿using ECommerceDemoCommon.DataProviders;
+using Microsoft.AspNetCore.Mvc;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ECommerceDataProvider.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class DataProviderController : ControllerBase
+    {
+        [HttpGet]
+        [Route("Entities")]
+        public async Task<IEnumerable<object>> Get(string entityType)
+        {
+            var dataProvider = new FileStringProvider(entityType);
+            var entities = await Task.Run(() => dataProvider.Get().ToList());
+            return entities;
+        }
+
+        [HttpGet]
+        public async Task<string> Get(string entityType, string id)
+        {
+            var dataProvider = new FileStringProvider(entityType);
+            string entity = await Task.Run(() => dataProvider.Get(id));
+            return entity;
+        }
+
+        [HttpPost]
+        public async Task Post(string entityType, string id, [FromBody] object entity)
```

Comment:
In a real-world application, it will be strongly restricted to have Object parameter in Action

Reply:
OK. The [FromBody] object parameter is removed. 
The posted/put json data are reading directly from the Request.Body (changes committed into main branch)